internal/internal_worker.go

```diff
@@ -874,66 +875,51 @@ func (aw *aggregatedWorker) Start() error {
 	return nil
 }
 
-var binaryChecksum string
-var binaryChecksumLock sync.Mutex
+var binaryChecksum atomic.Value
 
 // SetBinaryChecksum set binary checksum
 func SetBinaryChecksum(checksum string) {
-	binaryChecksumLock.Lock()
-	defer binaryChecksumLock.Unlock()
-
-	binaryChecksum = checksum
-}
-
-func initBinaryChecksum() error {
-	binaryChecksumLock.Lock()
-	defer binaryChecksumLock.Unlock()
-
-	return initBinaryChecksumLocked()
+	binaryChecksum.Store(checksum)
 }
 
-// callers MUST hold binaryChecksumLock before calling
-func initBinaryChecksumLocked() error {
-	if len(binaryChecksum) > 0 {
-		return nil
-	}
-
+func initBinaryChecksum() (string, error) {
 	exec, err := os.Executable()
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	f, err := os.Open(exec)
 	if err != nil {
-		return err
+		return "", err
 	}
 	defer func() {
 		_ = f.Close() // error is unimportant as it is read-only
 	}()
 
 	h := md5.New()
```

Comment:
hmmm.  tbh I forgot that this was the default behavior, since internally we report git-SHAs from an environment variable rather than checksumming the binary...

this might make an rwmutex better :|  checksumming big binaries can take time.  though at least it's only at startup.

Reply:
makes sense. let's lock around this part to avoid reading/hashing binary content multiple times